idaes/apps/grid_integration/multiperiod/price_taker_model.py

```diff
@@ -0,0 +1,1053 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+"""
+    Immediate ToDo list:
+        - Write methods to handle the representative day case
+            - automate cash flow expressions (likely use a 
+              weighted sum for representative days?)
+            - automate startup/shutdown constraints to
+              appropriately address the differences when
+              using representative days
+        - Add documentation to the IDAES readthedocs
+        - Add a warning when costs/revenues are given,
+          but are never used during cashflow construction
+"""
+import pandas as pd
+import numpy as np
+
+have_skl = True
+have_kn = True
+try:
+    from sklearn.cluster import KMeans
+except:
```

Comment:
This should use a more specific Exception type (ImportError?)

Reply:
Import errors are raised in specific methods since not all the methods require one or both of these imports (See L173 and L276)

Reply:
That is not the point I was getting at - it is generally bad practice to use a bare `except` as it can mask unexpected exceptions. Always specify the type of exception you want to handle, so that anything you did not expect gets raised as normal.

Reply:
Also, if different parts of the code have different dependencies, then that is a suggestion that they should be different modules to isolate the dependencies.